packages/cli/src/databases/migrations/mysqldb/1673268682475-DeleteExecutionsWithWorkflows.ts

```diff
@@ -0,0 +1,42 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+import { logMigrationEnd, logMigrationStart } from '@db/utils/migrationHelpers';
+import config from '@/config';
+
+export class DeleteExecutionsWithWorkflows1673268682475 implements MigrationInterface {
+	name = 'DeleteExecutionsWithWorkflows1673268682475';
+	public async up(queryRunner: QueryRunner): Promise<void> {
+		logMigrationStart(this.name);
+		const tablePrefix = config.getEnv('database.tablePrefix');
+
+		await queryRunner.query(`ALTER TABLE \`${tablePrefix}execution_entity\` MODIFY workflowId INT`);
+
+		await queryRunner.query(
+			`DELETE FROM \`${tablePrefix}execution_entity\`
+			 WHERE workflowId IS NOT NULL
+			 AND workflowId NOT IN (
+				 SELECT id FROM \`${tablePrefix}workflow_entity\`
```

Comment:
Subqueries are usually extremely slow (since you can reference columns from the outer query in the inner query, the subquery could end up being executed once for each row from the outer query.

I'd recommend either querying for workflow IDs separately or using a join here.

Reply:
done